pkg/proxy/backup.go

```diff
@@ -69,7 +74,72 @@ func (p *Proxy) SnapshotBackupStatus(ctx context.Context, req *rpc.EngineSnapsho
 
 	resp = &rpc.EngineSnapshotBackupStatusProxyResponse{}
 
-	// TODO
+	c, err := eclient.NewControllerClient(req.ProxyEngineRequest.Address)
+	if err != nil {
+		return nil, err
+	}
+	defer c.Close()
+
+	replicas, err := p.ReplicaList(ctx, req.ProxyEngineRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	replicaAddress := req.ReplicaAddress
+	if replicaAddress == "" {
+		// find a replica which has the corresponding backup
+		for _, r := range replicas.ReplicaList.Replicas {
+			mode := eptypes.GRPCReplicaModeToReplicaMode(r.Mode)
+			if mode != etypes.RW {
+				continue
+			}
+
+			cReplica, err := rclient.NewReplicaClient(r.Address.Address)
+			if err != nil {
+				logrus.Debugf("Failed to create replica client with %v", r.Address.Address)
+				continue
+			}
+
+			_, err = esync.FetchBackupStatus(cReplica, req.BackupName, r.Address.Address)
+			cReplica.Close()
+			if err == nil {
+				replicaAddress = r.Address.Address
+				break
+			}
+		}
+	}
+
+	if replicaAddress == "" {
+		return nil, errors.Errorf("failed to find a replica with backup %s", req.BackupName)
+	}
+
+	for _, r := range replicas.ReplicaList.Replicas {
+		if r.Address.Address != replicaAddress {
+			continue
+		}
+		mode := eptypes.GRPCReplicaModeToReplicaMode(r.Mode)
+		if mode != etypes.RW {
+			return nil, errors.Errorf("failed to get % v backup status on unknown replica %s", req.BackupName, replicaAddress)
+		}
+	}
+
+	cReplica, err := rclient.NewReplicaClient(replicaAddress)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create replica client with %v", replicaAddress)
+	}
+	defer cReplica.Close()
+
+	status, err := esync.FetchBackupStatus(cReplica, req.BackupName, replicaAddress)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We already got the backup status above (in the 1st for loop). Should we repeat it here?

Reply:
Removed.